packages/react-components/react-radio/src/components/RadioGroup/RadioGroup.types.ts

```diff
@@ -51,6 +51,11 @@ export type RadioGroupProps = Omit<ComponentProps<Partial<RadioGroupSlots>>, 'on
    * Require a selection in this group. Adds the `required` prop to all child Radio items.
    */
   required?: boolean;
+
+  /**
+   * Ref to the imperative API for RadioGroup. Allows setting focus to the selected radio item.
+   */
+  imperativeRef?: React.Ref<RadioGroupImperativeRef>;
```

Comment:
Is it possible to just use `ref` for this rather than introduce a new prop? I'm not very familiar with all the intricacies of refs but it seems like it would be nicer from an API and user point of view (if possible).

Reply:
The `ref` is an HTMLDivElement, and I don't want to try to mess around trying to make a proxy ref that overrides its `focus()` method. It might also be confusing since you're not actually focusing on the group's div; you're focusing on an item inside the group.

One possible alternative solution is to make the radiogroup div actually be focusable, and then listen for it to get focus and move focus to the appropriate place. That would let `ref.current.focus()` actually do what you're suggesting. However, it turns out to be tricky to get right... it's possible to do, but likely will have a bug tail. The main benefit of that approach is that it works with Tabster without requiring additional logic in Tabster.

Reply:
I'm not against helper imperative refs, this case in particular kinda makes sense as it's not straightfoward to get a handle on the checked radio. We could properly type an `imperativeRef` and document that. We'd have to be careful not to have too many imperativeRefs for straightforward use cases(i,e. imperativeRef.checkRadio) but that is a bit out of scope in this PR and discussion.

At the same time, I would push back on the proposal to extend the default `ref` because:

* Imperative refs do not proxy member attributes by design, I don't think we want to write overloads for every single HTMLDivElement property that is not function 😱
* We would extend the types of our forward ref elements - I think there's enough complexity in our compose types 🙈
* It's absolutely not standard practice in any other library
* If a user actually needs to actually make the radio group focusable and we overload the default `focus`- this is now impossible